packages/arb-avm-cpp/avm/include/avm/machinestate/machinestate.hpp

```diff
@@ -58,7 +58,9 @@ struct MachineState {
     BlockReason blockReason;
 
     MachineState();
-    MachineState(const std::vector<CodePoint>& code_, const value& static_val_);
+    MachineState(const std::vector<CodePoint>& code_,
+                 const value& static_val_,
+                 std::shared_ptr<TuplePool>& pool_);
```

Comment:
Pass this `shared_ptr` by value instead of reference to show that this is going to make a copy